server/src/operators/model_operator.rs

```diff
@@ -1030,3 +1058,494 @@ pub fn term_frequency(
+    } else {
+        message.clone()
+    };
+
+    let mut messages = vec![clipped_message.clone()];
+
+    if distance_phrase.is_some() {
+        let clipped_boost = if distance_phrase.as_ref().unwrap().phrase.len() > 7000 {
+            distance_phrase
+                .as_ref()
+                .unwrap()
+                .phrase
+                .chars()
+                .take(20000)
+                .collect()
+        } else {
+            distance_phrase.as_ref().unwrap().phrase.clone()
+        };
+        messages.push(clipped_boost);
+    }
+
+    let mut vectors = match embed_type {
+        "doc" => create_batch_embedding_call(messages, None),
+        "query" => create_batch_embedding_call(
+            vec![format!(
+                "{}{}",
+                dataset_config.EMBEDDING_QUERY_PREFIX, &clipped_message
+            )
+            .to_string()],
+            None,
+        ),
+        _ => create_batch_embedding_call(messages, None),
+    }
+    .await?;
+
+    if distance_phrase.is_some() {
+        let distance_factor = distance_phrase.unwrap().distance_factor;
+        let boost_vector = vectors.pop().unwrap();
+        let embedding_vector = vectors.pop().unwrap();
+
+        return Ok(embedding_vector
+            .iter()
+            .zip(boost_vector)
+            .map(|(vec_elem, boost_vec_elem)| vec_elem + distance_factor * boost_vec_elem)
+            .collect());
+    }
+
+    match vectors.first() {
+        Some(v) => Ok(v.clone()),
+        None => Err(ServiceError::InternalServerError(
+            "No dense embeddings returned from server".to_owned(),
+        )),
+    }
+}
+
+pub async fn create_embeddings_grpc(
+    content_and_boosts: Vec<(String, Option<DistancePhrase>)>,
+    _embed_type: &str,
+    _dataset_config: DatasetConfiguration,
+) -> Result<Vec<Vec<f32>>, ServiceError> {
+    let parent_span = sentry::configure_scope(|scope| scope.get_span());
+    let transaction: sentry::TransactionOrSpan = match &parent_span {
+        Some(parent) => parent
+            .start_child("create_embedding", "Create semantic dense embedding")
+            .into(),
+        None => {
+            let ctx = sentry::TransactionContext::new(
+                "create_embedding",
+                "Create semantic dense embedding",
+            );
+            sentry::start_transaction(ctx).into()
+        }
+    };
+    sentry::configure_scope(|scope| scope.set_span(Some(transaction.clone())));
+    let (contents, boosts): (Vec<_>, Vec<_>) = content_and_boosts.into_iter().unzip();
+    let (boost_indices, boost_phrases): (Vec<usize>, Vec<String>) = boosts
+        .clone()
+        .iter()
+        .enumerate()
+        .filter_map(|(index, boost)| {
+            boost
+                .clone()
+                .map(|distance_phrase| (index, distance_phrase.phrase))
+        })
+        .unzip();
+
+    let grpc_origin = get_env!("EMBEDDING_SERVER_GRPC_ORIGIN", "Grpc origin should be set");
```

Comment:
Remove `get_env!` and instead use `std::env::var()` since this is an optional parameter